pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -195,7 +195,7 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1alp
 		})
 		wakeUpControlPlane = g.Add(flow.Task{
 			Name:         "Waking up control plane to ensure proper cleanup of resources",
-			Fn:           flow.TaskFn(botanist.WakeUpControlPlane).DoIf((o.Shoot.Info.Status.IsHibernated != nil && *o.Shoot.Info.Status.IsHibernated || o.Shoot.Info.Status.IsHibernated == nil && o.Shoot.HibernationEnabled) && cleanupShootResources),
+			Fn:           flow.TaskFn(botanist.WakeUpControlPlane).DoIf(((o.Shoot.Info.Status.IsHibernated != nil && *o.Shoot.Info.Status.IsHibernated) || ((o.Shoot.Info.Status.IsHibernated == nil || !*o.Shoot.Info.Status.IsHibernated) && o.Shoot.HibernationEnabled)) && cleanupShootResources),
```

Comment:
```suggestion
			Fn:           flow.TaskFn(botanist.WakeUpControlPlane).DoIf(((o.Shoot.Info.Status.IsHibernated != nil && *o.Shoot.Info.Status.IsHibernated) || o.Shoot.HibernationEnabled) && cleanupShootResources),
```

Should have the same effect, shouldn't it?

Reply:
Yep, the look the same.